Mark-In/Sources/App/DIContainer.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  DIContainer.swift
+//  Mark-In
+//
+//  Created by 이정동 on 5/5/25.
+//
+
+import Foundation
+
+import LinkMetadataKit
+import LinkMetadataKitInterface
+
+// TODO: 코어 모듈로 이전
+final class DIContainer {
+  static let shared = DIContainer()
+  private var dependencies: [String: Any] = [:]
+  
+  private init() {}
+  
+  func register<T>(_ dependency: T) {
+    let key = String(describing: T.self)
+    dependencies[key] = dependency
+  }
+  
+  func resolve<T>() -> T {
+    let key = String(describing: T.self)
+    let dependency = dependencies[key]
+    
+    guard let dependency = dependency as? T else {
+      fatalError("\(key)는 register되지 않았어어요. resolve 부르기전에 register 해주세요")
+    }
+    
+    return dependency
+  }
+}
+
+extension DIContainer {
+  func registerDependencies() {
+    
+    /// Core
+    let linkMetadataProvider: LinkMetadataProvider = LinkMetadataProviderImpl()
+    let authSessionManager: AuthManager = AuthManagerImpl()
```

Comment:
```suggestion
    let authManager: AuthManager = AuthManagerImpl()
```
이름 동일하게 authManager로 하는게 좋을 것 같아요 세션이 붙으면 혼란이 있을 것 같아서요~